### PR TITLE
Reusable Material Pipeline

### PR DESCRIPTION
**Pull request by VVishion**
# Objective

The `MaterialPipeline` cannot be integrated into other pipelines like the `MeshPipeline` can.

## Solution

Implement `Clone` for `MaterialPipeline`. Expose systems and resources part of the `MaterialPlugin` to allow custom assembly - especially combining existing systems and resources with a custom `queue_material_meshes` system.

---

### Added

- `Clone` impl for `MaterialPipeline`

### Changed

- `ExtractedMaterials`, `extract_materials` and `prepare_materials` are now public

### CODE REVIEW

File: crates/bevy_pbr/src/material.rs
Comment (VVishion) [inline crates/bevy_pbr/src/material.rs:383]: This is just a change of indentation.

**Review by alice-i-cecile — APPROVED** [review]
Wow, shockingly simple. This has my approval.

**Review by kurtkuehnert — APPROVED** [review]
I have run into this issue before. I really like this simple solution.  Also removing the double indentation is nice.

This PR is closely related to #6653, which implements `Clone` for all pipelines.

**Comment by alice-i-cecile** [comment]
@VVishion once this is rebased or the merge conflicts resolved I'll merge this.

**Comment by kurtkuehnert** [comment]
Can we get this merged for 0.10?
I can update this PR if necessary.

**Comment by alice-i-cecile** [comment]
@kurtkuehnert can you open a new PR to adopt this? I'd like to get this in too.

**Comment by alice-i-cecile** [comment]
Adopted in #7548 :)

**Pull request closed without merging**
